src/shop-payment-request.html

```diff
@@ -0,0 +1,254 @@
+
+<dom-module id="shop-payment-request">
+  <script>
+
+    Polymer({
+      is: 'shop-payment-request',
+
+      properties: {
+        currency: {
+          type: String,
+          value: 'USD'
+        },
+
+        supportedMethods: {
+          type: Array,
+          value: function() { return ["visa", "mastercard", "amex", "discover", "maestro", "diners", "jcb", "unionpay", "bitcoin"]; }
+        },
+
+        requestPayerPhone: {
+          type: Boolean,
+          value: false
+        },
+
+        requestPayerEmail: {
+          type: Boolean,
+          value: false
+        },
+
+        androidPay: {
+          type: Object,
+          value: function() { return {}; }
+        },
+
+        summaryLabel: {
+          type: String,
+          value: "Total"
+        },
+
+        _paymentTypesSupported: {
+          computed: '_computePaymentTypesSupported(supportedMethods, androidPay)'
+        }
+      },
+
+      request: function(itemDetails, totalCost, currency, clearCartOnSuccess) {
+        if (!'PaymentRequest' in window) {
+          this._redirectToCheckout();
+        }
+
+        this.currency = currency;
+
+        var details = {
+          total: { label: this.summaryLabel, amount:
+            { currency: this.currency, value: totalCost.toFixed(2) }
+          },
+          displayItems: itemDetails
+        };
+
+        // Collect a shipping address, and honor client's other options.
+        var options = {
+          requestShipping: true,
+          requestPayerEmail: this.requestPayerEmail,
+          requestPayerPhone: this.requestPayerPhone
+        };
+
+        try {
+          var request = new PaymentRequest(this._paymentTypesSupported, details, options);
+
+          var that = this;
+
+          // Handle changing shipping options
+          request.addEventListener('shippingoptionchange', function(e) {
+              e.updateWith(new Promise(function(resolve) {
+                  that._updateShippingOptions(details, request.shippingOption, resolve);
+              }));
+          });
+
+          // Handle selecting/changing the shipping address
+          request.addEventListener('shippingaddresschange', function(e) {
+            e.updateWith(new Promise(function(resolve) {
+                that._updateShippingAddress(details, request.shippingAddress, resolve);
+            }));
+          });
+
+          request
+            .show()
+            .then(function(instrumentResponse) {
+              that.fire('process-payment', {
+                'instrument': instrumentResponse,
+                'cartBuy': clearCartOnSuccess
+              });
+            })
+            .catch(function(err) {
+              if (err === 'Request cancelled') {
+                that.fire('error', {error_code: 0, message: err});
+              } else if (err.name == 'NotSupportedError') {
+                that.fire('error', {error_code: 1, message: err.message});
```

Comment:
nit: `1` could be a constant. e.g. `Polymer.ShopPaymentRequest.NOT_SUPPORTED = 1;` that way the user could check for them in the error handler.  

A proposal:
```js
Polymer.ShopPaymentRequest =  Polymer({ is: 'shop-payment-request',  ... });
Polymer.ShopPaymentRequest.NOT_SUPPORTED = 1;
...
...
```